GetIntoTeachingApi/Controllers/TeacherTrainingAdviser/CandidatesController.cs

```diff
@@ -40,11 +44,19 @@ public CandidatesController(ILogger<CandidatesController> logger)
             OperationId = "GetExistingTeacherTrainingAdviserCandidate",
             Tags = new[] { "Teacher Training Adviser" }
         )]
+        [ProducesResponseType(401)]
         public IActionResult Get(
-            [FromRoute, SwaggerParameter("Access token (PIN code).", Required = true)] string accessToken, 
+            [FromRoute, SwaggerParameter("Access token (PIN code).", Required = true)] string accessToken,
+            [FromHeader(Name = "Candidate-Email"), SwaggerParameter("Candidate email address.", Required = true)] string email,
             [FromHeader(Name = "Authorization"), SwaggerParameter("Bearer <sharedSecret>.", Required = true)] string sharedSecret
         )
         {
+            var challenge = new CandidateAccessTokenChallenge { Token = accessToken, SharedSecret = sharedSecret, Email = email };

```

Comment:
Bear with me, I could be reading this wrong (first time I've really looked at C#), _but_ this looks wrong - there's should be two secrets here

1. Bearer token in header used for server <--> client auth - _ideally we can support multiple for different clients, comma separated maybe?_
2. Private internal secret only available to server, this gets combined with email for generating TOTP

At present the API client has all the information to generate valid TOTP codes itself?

Reply:
Yeah you're right in that I've used the same token for the shared secret and the TOTP secret; I didn't think the client being able to generate valid codes would be an issue as I believe this is the case with the TOTP algorithm anyway - but you make a good point that we're likely going to have multiple clients in the future so that should be separated 👍